docs/advanced-features.md

```diff
@@ -312,3 +312,20 @@ releases:
   - chart: oci://my-oci-registry/helm-repo/envoy
     version: 1.5
 ```
+
+### Lockfile per environment
+
+In some cases it can be handy for CI/CD pipelines to be able to roll out updates gradually for environments, such as staging and production while using the same
+set of charts. This can be achieved by using `lockFile` in combination with environments, such as:
+
+```yaml
+lockFile: .helmfile.{{ .Environment.Name}}.lock
+
+environments:
+  staging:
+  production
+
+releases:
+- name: myapp
+  chart: charts/myapp
+```
```

Comment:
Thanks for the PR!
One thing I noticed is that you should never use a helmfile template within the same fragment that defines `environments`. See https://github.com/helmfile/helmfile/blob/main/docs/proposals/towards-1.0.md#the-changes-in-10
This should ideally be written something like the below:

```suggestion
```yaml
environments:
  staging:
  production

---
lockFile: .helmfile.{{ .Environment.Name}}.lock

releases:
- name: myapp
  chart: charts/myapp
```
```